src/main/java/seedu/system/model/person/CustomDate.java

```diff
@@ -15,11 +15,14 @@
             "Date should follow the following format DD/MM/YYYY.";
     public static final String DATE_FORMAT = "dd/MM/yyyy";
     public final String date;
+    public final Date dateObj;
 
-    public CustomDate(String date) {
+    public CustomDate(String date) throws ParseException {
         requireNonNull(date);
         checkArgument(isValidDate(date.trim()), MESSAGE_CONSTRAINTS);
         this.date = date.trim();
+        SimpleDateFormat format = new SimpleDateFormat(DATE_FORMAT);
+        dateObj = format.parse(date);
     }
```

Comment:
so this constructor is good.

but what you want to do for the tests is that you want to create another CustomDate constructor:

public CustomDate(Date dateObj) {
        requireNonNull(dateObj);
        // rest of code
}

that doesnt throw a parse exception, so that you dont have to trhow the parseexceptions all over the code base especially in the tests and sample data util